patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch

```diff
@@ -0,0 +1,77 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 9 Dec 2022 13:08:16 +0100
+Subject: [PATCH] Don't show particles when splash event are canceled
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/projectile/ThrownPotion.java b/src/main/java/net/minecraft/world/entity/projectile/ThrownPotion.java
+index 934b9e76d9d7d4170ac9205622bc87975de73541..26e879ffcbfb7c8cf0fa5e79d461127720828de5 100644
+--- a/src/main/java/net/minecraft/world/entity/projectile/ThrownPotion.java
++++ b/src/main/java/net/minecraft/world/entity/projectile/ThrownPotion.java
+@@ -110,26 +110,28 @@ public class ThrownPotion extends ThrowableItemProjectile implements ItemSupplie
+             Potion potionregistry = PotionUtils.getPotion(itemstack);
+             List<MobEffectInstance> list = PotionUtils.getMobEffects(itemstack);
+             boolean flag = potionregistry == Potions.WATER && list.isEmpty();
++            boolean showParticles = true; // Paper
+ 
+             if (flag) {
+-                this.applyWater();
++                showParticles = this.applyWater(); // Paper
+             } else if (true || !list.isEmpty()) { // CraftBukkit - Call event even if no effects to apply
+                 if (this.isLingering()) {
+-                    this.makeAreaOfEffectCloud(itemstack, potionregistry);
++                    showParticles = this.makeAreaOfEffectCloud(itemstack, potionregistry); // Paper
+                 } else {
+-                    this.applySplash(list, hitResult != null && hitResult.getType() == HitResult.Type.ENTITY ? ((EntityHitResult) hitResult).getEntity() : null); // Paper - nullable hitResult
++                    showParticles = this.applySplash(list, hitResult != null && hitResult.getType() == HitResult.Type.ENTITY ? ((EntityHitResult) hitResult).getEntity() : null); // Paper - nullable hitResult / cancel particle when event is canceled
+                 }
+             }
+ 
+             int i = potionregistry.hasInstantEffects() ? 2007 : 2002;
+ 
++            if (showParticles) // Paper
```

Comment:
Use brackets to improve readability and increase chance of conflict if something changes.